vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -0,0 +1,466 @@
+    failed_when:  running_upgraded_containers.stdout|int < 8
+
+# ---------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
+#                       Upgrade K8S version and boot-image                              |
+# ---------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
+  - name: get cluster uid
+    shell: |
+            kubectl get clusters {{ CLUSTER_NAME }}  -n {{NAMESPACE}} -o json | jq '.metadata.uid' |   cut -f2 -d\"
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: CLSTR_UID
+
+  - name: Genenrate controlplane Metal3MachineTemplate
+    vars:
+       CLUSTER_UID: "{{ CLSTR_UID.stdout }}"
+       M3MT_NAME: "{{CLUSTER_NAME}}-new-controlplane-image"
+       DATA_TEMPLATE_NAME: "{{CLUSTER_NAME}}-controlplane-template"
+    template:
+      src: Metal3MachineTemplate.yml
+      dest: /tmp/cp_new_image.yaml
+
+  - name: Genenrate worker Metal3MachineTemplate
+    vars:
+      CLUSTER_UID: "{{ CLSTR_UID.stdout_lines[0] }}"
+      M3MT_NAME: "{{CLUSTER_NAME}}-new-workers-image"
+      DATA_TEMPLATE_NAME: "{{CLUSTER_NAME}}-workers-template"
+    template:
+      src: Metal3MachineTemplate.yml
+      dest: /tmp/wr_new_image.yaml
+
+  - name: Create controlplane and worker Metal3MachineTemplates
+    shell: |
+        kubectl apply -f /tmp/cp_new_image.yaml
+        kubectl apply -f /tmp/wr_new_image.yaml
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Update boot-disk and kubernetes versions of controlplane nodes
+    shell: |
+            kubectl get kubeadmcontrolplane -n {{NAMESPACE}} {{ CLUSTER_NAME }} -o json |
+            jq '.spec.infrastructureTemplate.name="{{CLUSTER_NAME}}-new-controlplane-image" |
+               .spec.version="{{UPGRADED_K8S_VERSION}}"'|
+                kubectl apply -f-
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Verify that controlplane nodes using the new node image
+    shell: |
+            kubectl get bmh -n {{NAMESPACE}} |
+            grep -i provisioned | grep -c 'new-controlplane-image'
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    retries: 200
+    delay: 20
+    register: new_image_cp_nodes
+    until: new_image_cp_nodes.stdout|int == 3
+    failed_when: new_image_cp_nodes.stdout|int != 3
+
+  - name: Wait for old etcd instance to leave the new etcd-cluster
+    pause:
+      minutes: 2
+
+  - name: Untaint all CP nodes after upgrade of controlplane nodes
+    shell: |
+        kubectl taint nodes --all node-role.kubernetes.io/master-
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    ignore_errors: yes
+
+  - name: Verify that the old controlplane node has left the cluster
+    shell: |
+            kubectl get bmh -n {{NAMESPACE}} | grep -i provisioned | grep -c 'test1-controlplane-'
```

Comment:
test1 should be "{{ CLUSTER_NAME }}"??

Reply:
changed

Reply:
But, line 6 above, It does not refer to a cluster. It is a machinedeployment name (hard coded as well)